pandas/_libs/interval.pyx

```diff
@@ -81,26 +81,45 @@ cdef class Interval(IntervalMixin):
 
     Parameters
     ----------
-    left : value
-        Left bound for the interval
-    right : value
-        Right bound for the interval
+    left : orderable object
+        Left bound for the interval.
+    right : orderable object
+        Right bound for the interval.
     closed : {'left', 'right', 'both', 'neither'}, default 'right'
         Whether the interval is closed on the left-side, right-side, both or
-        neither
+        neither.
+    
+    Notes
+    -----
+    You must be able to compare the parameters **left** and **right**,
+    and they must satisfy **left <= right**. 
 
     Examples
     --------
+    The examples below show how you can build Intervals of various types
+    of elements. In particular a numeric interval, a time interval and
+    finally a string interval. Given an Interval you can check whether 
+    an element belongs to it.
```

Comment:
In general, I think it can be easier to follow to put some of those explanations between the examples, instead of first all explanation and then all examples.